### DISCUSSION TITLE
Voices management

### DISCUSSION THREAD

**Discussion by GillesLACAUD** [marcel-licence/esp32_basic_synth#10] [Ideas]
For now when you hit a new key and there is no more free voice you just return a null in you function. You can insert a "steal note" algorithm so the first key will be off and the new key on

**Comment by marcel-licence** [comment]
Good idea. I also noticed that it is problematic when hitting new keys before releasing the old one. Is there a desired priority which key should be "stolen"?

**Comment by GillesLACAUD** [comment]
Imagine you have a 4 voices poly synth and these notes :
**C2 C4 E4 G4** and you want to add the B4...the B4 must stop the first note C2 then the B4 is playing. I have solve this with the active boolean, change it to an uint8_t and write a rank on it. **C2(1) C4(2) E4(3) G4(4)** with the B4 the first rand is off and we decrease all the upper ranks so **C4(1) E4(2) G4(3) B4(4).** Let suppose now E4 is off...so we must decrease all the upper ranks: **C4(1) G4(2) B4(3).**
